src/Workspaces/Core/Portable/Workspace/Solution/ProjectId.cs

```diff
@@ -28,8 +28,12 @@ private ProjectId(string debugName)
             _debugName = debugName;
         }
 
-        internal ProjectId(Guid guid, string debugName)
+        public ProjectId(Guid guid, string debugName = null)
```

Comment:
@Pilchie I agree it should be one or the other.  I dislike CreateFromSerialized because it is naming this specific use case vs describing the operation.  In general I think overloaded constructors are better than multiple named Create methods (CreateA, CreateB).